src/rgw/rgw_rest_swift.cc

```diff
@@ -1205,17 +1205,20 @@ int RGWGetObj_ObjStore_SWIFT::send_response_data(bufferlist& bl,
 
   set_req_state_err(s, (partial_content && !op_ret) ? STATUS_PARTIAL_CONTENT
 		    : op_ret);
+
   dump_errno(s);
-  if (s->err.is_err()) {
-    end_header(s, NULL);
-    return 0;
-  }
 
   if (range_str) {
     dump_range(s, ofs, end, s->obj_size);
   }
 
   dump_content_length(s, total_len);
+
```

Comment:
In case of an error the content length of the object doesn't matter, 
it should be the content length of the error message (some error message have body)